### PR TITLE
feat: add mrr evaluator

### PR DESCRIPTION
**Pull request by bhavsarpratik**

### CODE REVIEW

File: evaluators/rank/MeanReciprocalRankEvaluator/__init__.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    def evaluate(self, actual: Sequence[Sequence[Union[str,int]]], desired: Sequence[Sequence[Union[str,int]]], *args, **kwargs) -> float:
+        """
+        :param actual: should be a sequence of sequence of document IDs
+        :param desired: should be a sequence of sequence of document IDs
+        :return gives mean reciprocal rank score
+        """
+        import numpy as np
+        if len(actual)==0 or len(desired)==0:
+            return 0.0
+        return float(np.mean([self._evaluate(actual_i, desired_i) for actual_i, desired_i in zip(actual, desired)]))
```
Comment (JoanFM) [inline evaluators/rank/MeanReciprocalRankEvaluator/__init__.py:35]: Why is this cast needed?

File: evaluators/rank/MeanReciprocalRankEvaluator/tests/test_reciprocalrankevaluator.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    ([],[], 0.0),
+    ([[1, 2, 3, 4],[]], [[], []], 0.0),
+    ([[], []], [[1, 2, 3, 4],[]], 0.0),
+    ([[1, 2, 3, 4]],[[1, 2, 3, 4]], 1.0),
+    ([[1, 2, 3, 4]],[[11, 1, 2, 3]], 0.0),
+    ([[4, 2, 3, 1]],[[1, 2, 3, 4]], 0.0),
+    ([[1, 2, 3, 4], [1, 2, 3, 4]], [[1, 2, 3, 4], [2, 1, 3, 4]], (1+0.5)/2)
+]
+)
+def test_reciprocalrankevaluator(actual, desired, score):
```
Comment (JoanFM) [inline evaluators/rank/MeanReciprocalRankEvaluator/tests/test_reciprocalrankevaluator.py:17]: Change name of test and test file to test_mean...

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: evaluators/rank/MeanReciprocalRankEvaluator/__init__.py (hunk 1 above)
Comment (bhavsarpratik) [reply evaluators/rank/MeanReciprocalRankEvaluator/__init__.py:35]: np.mean returns <class 'numpy.float64'>

File: evaluators/rank/MeanReciprocalRankEvaluator/__init__.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        except:
+            return 0.0
+
+    def evaluate(self, actual: Sequence[Sequence[Union[str,int]]], desired: Sequence[Sequence[Union[str,int]]], *args, **kwargs) -> float:
+        """
+        :param actual: should be a sequence of sequence of document IDs
+        :param desired: should be a sequence of sequence of document IDs
+        :return gives mean reciprocal rank score
+        """
+        import numpy as np
```
Comment (JoanFM) [inline evaluators/rank/MeanReciprocalRankEvaluator/__init__.py:32]: Numpy import can be at the top.

Comment (bhavsarpratik) [reply evaluators/rank/MeanReciprocalRankEvaluator/__init__.py:32]: Which imports can be at the top? Can torch be at the top?

**Review by JoanFM — APPROVED** [review]

**Pull request merged**